src/hooks/useInterval.ts

```diff
@@ -1,27 +1,55 @@
-import { useEffect, useRef } from 'react'
+import { useEffect, useRef, useState, useCallback } from 'react'
+import isUndefinedOrNull from 'utils/isUndefinedOrNull'
+import useLastUpdated from './useLastUpdated'
 
-export default function useInterval(callback: () => void, delay: null | number, leading = true) {
-  const savedCallback = useRef<() => void>()
+export default function useInterval(
+  callback: () => void,
+  delay: undefined | null | number,
+  leading = true,
+  initiateUpdate = true,
```

Comment:
what's the `initiateUpdate` here for?

Reply:
This is for to run interval/timeout on the background but not calling callback function (which is list updater in our example) when time has come but user is not in swap / liquidity pages. Reason for doing that to avoid calling list updater everytime user access those pages.